crates/runtime/src/lib.rs

```diff
@@ -356,9 +357,51 @@ impl_runtime_apis! {
             Some((tx_type, events))
         }
 
+        fn get_tx_messages_to_l1(tx_hash: TransactionHash) -> Vec<MessageToL1> {
+            Starknet::tx_messages(tx_hash)
+        }
+
         fn get_tx_execution_outcome(tx_hash: TransactionHash) -> Option<Vec<u8>> {
-           Starknet::tx_revert_error(tx_hash).map(|s| s.into_bytes())
+            Starknet::tx_revert_error(tx_hash).map(|s| s.into_bytes())
+        }
+
+        fn get_starknet_messages(block_extrinsics: Vec<<Block as BlockT>::Extrinsic>, chain_id: Felt252Wrapper) -> (Vec<MessageL2ToL1>, Vec<MessageL1ToL2>) {
```

Comment:
`extract_starknet_messages` rather than `get`?

Reply:
This is also not my code...